Build failure: rPackages.tesseract

### Steps To Reproduce

Steps to reproduce the behavior:

Looks like rPackages.tesseract is failing to build.

```
nix build -f default.nix rPackages.tesseract
```

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/c7i21b5ild18y42abdsr2a1y41vv2v73-tesseract_5.1.0.tar.gz
source root is tesseract
setting SOURCE_DATE_EPOCH to timestamp 1653821402 of file tesseract/MD5
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
@nix { "action": "setPhase", "phase": "installPhase" }
installing
* installing *source* package 'tesseract' ...
** package 'tesseract' successfully unpacked and MD5 sums checked
** using staged installation
Found pkg-config cflags and libs!
Using PKG_CFLAGS=-I/nix/store/2b3samg5y4vqla4amja15ysbwnhm38jy-tesseract-5.3.3/include
Using PKG_LIBS=-L/nix/store/2b3samg5y4vqla4amja15ysbwnhm38jy-tesseract-5.3.3/lib -ltesseract
Using CXX11CPP: /nix/store/zlzz2z48s7ry0hkl55xiqp5a73b4mzrg-gcc-wrapper-12.3.0/bin/c++ -E -std=gnu++11
--------------------------- [ANTICONF] --------------------------------
Configuration failed to find 'tesseract' system library. Try installing:
 * deb: libtesseract-dev libleptonica-dev (Debian, Ubuntu, etc)
 * rpm: tesseract-devel leptonica-devel (Fedora, CentOS, RHEL)
 * brew: tesseract (Mac OSX)
If tesseract is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a tesseract.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
-------------------------- [ERROR MESSAGE] ---------------------------
tools/test.cpp:2:10: fatal error: allheaders.h: No such file or directory
    2 | #include <allheaders.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
--------------------------------------------------------------------
ERROR: configuration failed for package 'tesseract'
* removing '/nix/store/ajs8ypr7xis62wc65ia6igrhr5yrk7hk-r-tesseract-5.1.0/library/tesseract'
```

### Additional context

Tested on x86_64-linux

### Notify maintainers

@jbedo

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -1387,6 +1387,13 @@ let
       '';
     });
 
+    tesseract = old.tesseract.overrideAttrs (_: {
+      preConfigure = ''
+        substituteInPlace configure \
+          --replace 'PKG_CONFIG_NAME="tesseract"' 'PKG_CONFIG_NAME="tesseract lept"'
+      '';
+    });
+
     ijtiff = old.ijtiff.overrideAttrs (_: {
       preConfigure = ''
         patchShebangs configure

